### PR TITLE
DolphinQt2: Show cached games before checking whether they exist on disk

### PR DESCRIPTION
**Pull request by JosJuice**
DolphinWX already has this improvement in startup time, and it matters a lot when you have a large game list.

### CODE REVIEW

File: Source/Core/DolphinQt2/GameList/GameTracker.h [hunk 1]
```diff
@@ -39,7 +40,7 @@ class GameTracker final : public QFileSystemWatcher
 
 signals:
   void GameLoaded(std::shared_ptr<const UICommon::GameFile> game);
-  void GameRemoved(const QString& path);
+  void GameRemoved(const std::string& path);
```
Comment (spycrab) [inline Source/Core/DolphinQt2/GameList/GameTracker.h]: ``std::string`` will need to be declared as a metatype (``Q_DECLARE_METATYPE``) if it isn't somewhere else already.

Comment (JosJuice) [reply Source/Core/DolphinQt2/GameList/GameTracker.h]: Done.

File: Source/Core/UICommon/GameFileCache.cpp [hunk 2]
```diff
@@ -102,21 +105,27 @@ bool GameFileCache::Update(const std::vector<std::string>& all_game_paths)
...
+        m_cached_files.pop_back();
       }
     }
-    m_cached_files.erase(it, m_cached_files.end());
   }
 
   // Now that the previous loop has run, game_paths only contains paths that
   // aren't in m_cached_files, so we simply add all of them to m_cached_files.
-  for (const auto& path : game_paths)
+  for (const std::string& path : game_paths)
```
Comment (spycrab) [inline Source/Core/UICommon/GameFileCache.cpp]: Why change it here when you kept ``auto`` in the other loops?

Comment (JosJuice) [reply Source/Core/UICommon/GameFileCache.cpp]: Which other loops? I can only find one loop in this file that uses auto, and that auto represents a longer type than std::string. Though I guess I could change that one too...

**Pull request merged**